src/lein_monolith/dependency.clj

```diff
@@ -176,8 +244,20 @@
      ; should appear *later* in the sequence.
      (let [roots (apply set/difference (set (keys m)) (map set (vals m)))]
        (when (empty? roots)
-         (throw (ex-info "Cannot sort the keys in the given map, cycle detected!"
-                         {:input m})))
+         (let [sort-cycles #(->> %
+                                 (group-by count)
+                                 (into (sorted-map))
+                                 vals
+                                 (mapcat identity))
```

Comment:
Isn't this the same as `(sort-by count %)`?

Reply:
Yes, thanks fixed.